wtforms/fields/core.py

```diff
@@ -456,6 +456,10 @@ def iter_choices(self):
 
     def process_data(self, value):
         try:
+            # protect against coercing None,
+            # such as in text_type(None) -> "None"
+            if value is None:
+                raise ValueError()
             self.data = self.coerce(value)
```

Comment:
I don't like how this is raising an error just to be caught, would it not be better to do
```python
try:
    self.data = None if value is None else self.coerce(value)
except (ValueError, TypeError):
    self.data = None
```

Reply:
I have no strong opinion on this.

Reply:
In that case I'll change it when I merge it since I have to do it manually